icu4j/main/core/src/test/java/com/ibm/icu/dev/test/format/JavaTimeFormatTest.java

```diff
@@ -66,6 +68,75 @@ public void testLocalDateTimeFormatting() {
     }
 
+    @Test
+    public void testDayOfWeekFormatting() {
+        // Quick sampling
+        DateFormat df = DateFormat.getInstanceForSkeleton("EEEE", Locale.FRANCE);
+        assertEquals("", "lundi", df.format(DayOfWeek.MONDAY));
+        assertEquals("", "mardi", df.format(DayOfWeek.TUESDAY));
+        assertEquals("", "jeudi", df.format(DayOfWeek.THURSDAY));
+        assertEquals("", "samedi", df.format(DayOfWeek.SATURDAY));
+        assertEquals("", "dimanche", df.format(DayOfWeek.SUNDAY));
+
+        // Comprehensive, testing all day-of-week at all widths
+        int[] widths = {
+                DateFormatSymbols.NARROW,
+                DateFormatSymbols.SHORT,
+                DateFormatSymbols.ABBREVIATED,
+                DateFormatSymbols.WIDE
+        };
+        String[] widthSkeletons = { "EEEEE", "EEEEEE", "EEE", "EEEE" };
+        assertEquals("same size", widths.length, widthSkeletons.length);
+
+        Locale locale = Locale.US;
+        DateFormatSymbols dfs = DateFormatSymbols.getInstance(locale);
+        for (int i = 0; i < widths.length; i++) {
+            String[] expected = dfs.getWeekdays(DateFormatSymbols.FORMAT, widths[i]);
+            df = DateFormat.getInstanceForSkeleton(widthSkeletons[i], locale);
+            for (DayOfWeek dow : DayOfWeek.values()) {
+                int index = dow.getValue();
+                // DayOfWeek is 1 based and starting on Monday.
+                // DateFormatSymbols day of week starts on Sunday,
+                // and the first element in the array empty: [ "", "Su", "Mo", ... "Sa" ]
+                index = index % 7 + 1; // convert the index
+                assertEquals("java.time.DayOfWeek", expected[index], df.format(dow));
+            }
+        }
+    }
+
+    @Test
+    public void testMonthFormatting() {
+        // Quick sampling
+        DateFormat df = DateFormat.getInstanceForSkeleton("MMMM", Locale.FRANCE);
+        assertEquals("", "janvier", df.format(Month.JANUARY));
+        assertEquals("", "mars", df.format(Month.MARCH));
+        assertEquals("", "mai", df.format(Month.MAY));
+        assertEquals("", "juillet", df.format(Month.JULY));
+        assertEquals("", "septembre", df.format(Month.SEPTEMBER));
+        assertEquals("", "novembre", df.format(Month.NOVEMBER));
+        assertEquals("", "décembre", df.format(Month.DECEMBER));
+
+        // Comprehensive, testing all months at all widths
+        int[] widths = {
+                DateFormatSymbols.NARROW,
+                DateFormatSymbols.ABBREVIATED,
+                DateFormatSymbols.WIDE
+        };
+        String[] widthSkeletons = { "MMMMM", "MMM", "MMMM" };
+        assertEquals("same size", widths.length, widthSkeletons.length);
+
+        Locale locale = Locale.US;
+        DateFormatSymbols dfs = DateFormatSymbols.getInstance(locale);
+        for (int i = 0; i < widths.length; i++) {
+            String[] expected = dfs.getMonths(DateFormatSymbols.FORMAT, widths[i]);
```

Comment:
Does the month symbols array also start with an empty string in the first element?

Reply:
No, they start at zero, with January.

```java
DateFormatSymbols dfs = DateFormatSymbols.getInstance(Locale.US);
System.out.println(Arrays.toString(dfs.getMonths()));
System.out.println(Arrays.toString(dfs.getWeekdays()));
```
Result:
```
[January, February, March, April, May, June, July, August, September, October, November, December]
[, Sunday, Monday, Tuesday, Wednesday, Thursday, Friday, Saturday]
```